### PR TITLE
Gio Update

### PR DESCRIPTION
**Pull request by oshorefueled**
### Resolves
This PR updates the Gio version. It includes password masking feature on editor and fixes the issue with UI lag when resizing the window on MacOS. It also reverts the decredmaterial custom theme to be based off the gioui material theme.

### CODE REVIEW

File: ui/window.go [hunk 1]
```diff
@@ -20,7 +23,9 @@ import (
 type Window struct {
 	window *app.Window
 	theme  *decredmaterial.Theme
-	gtx    *layout.Context
+	// gtx    *layout.Context
```
Comment (Sirmorrison) [inline ui/window.go]: If no longer needed lets remove the commented lines of codes.

File: main.go [hunk 2]
```diff
@@ -54,10 +51,10 @@ func main() {
 			log.Warn(err)
 		}
 		if fnt != nil {
-			font.Register(text.Font{}, fnt)
+			// font.Register(text.Font{}, fnt)
```
Comment (Sirmorrison) [inline main.go]: If no longer needed lets remove the commented lines of codes.

File: ui/recieve_page.go [hunk 3]
```diff
@@ -20,27 +20,26 @@ const PageReceive = "receive"
...
 
 	isNewAddr, isInfo bool
 	addrs             string
 
-	newAddrBtn, minInfo                                 decredmaterial.Button
-	copyBtn, infoBtn, moreBtn                           decredmaterial.IconButton
-	copyBtnW, infoBtnW, moreBtnW, minInfoW, newAddrBtnW widget.Button
+	newAddrBtn, minInfo       decredmaterial.Button
+	copyBtn, infoBtn, moreBtn decredmaterial.IconButton
+	// copyBtnW, infoBtnW, moreBtnW, minInfoW, newAddrBtnW widget.Clickable
```
Comment (Sirmorrison) [inline ui/recieve_page.go:29]: If no longer needed lets remove the commented lines of codes.

File: ui/page.go [hunk 4]
```diff
@@ -28,44 +30,49 @@ type pageCommon struct {
...
-	theme           *decredmaterial.Theme
-	icons           pageIcons
-	page            *string
-	navTab          *decredmaterial.Tabs
-	walletsTab      *decredmaterial.Tabs
-	accountsTab     *decredmaterial.Tabs
-	errorChannels   map[string]chan error
-	keyEvents       chan *key.Event
-	states          *states
+	// gtx             layout.Context
```
Comment (Sirmorrison) [inline ui/page.go]: If no longer needed lets remove the commented lines of codes.

**Review by Sirmorrison — COMMENTED** [review]
Nice work so for with the update to the gioui package. 

I especially like the widget extension you did as this resulted in a major code clean up.

I think whats left with this PR is to remove the commented codes that are no longer needed and we can have this PR merged for further work.

**Review by beansgum — APPROVED** [review]

**Pull request merged**